lib/logproto/logproto-proxied-text-server.c

```diff
@@ -189,6 +189,34 @@ _log_proto_proxied_text_server_add_aux_data(LogProtoProxiedTextServer *self, Log
   return;
 }
 
+static inline LogProtoStatus
+_fetch_into_proxy_buffer(LogProtoProxiedTextServer *self, gsize *hdr_len)
+{
+  for(gint i = 0; i < PROXY_PROTO_HDR_MAX_LEN_RFC; i++)
+    {
+      gssize rc = log_transport_read(self->super.super.super.transport, &(self->v1_proxy_header_buff[i]), sizeof(gchar),
```

Comment:
Protos are non-blocking, `EAGAIN` has to be handled properly. which means that the for loop (in the current form) won't work either, because `_fetch_into_proxy_buffer()` will be called again in that case, and we should be able to continue reading the header where we left off.

When detecting `EAGAIN`, `LPS_AGAIN` must be returned from the proto implementation.
We also need to implement the `prepare()` virtual method, which should the IO direction to `G_IO_IN` and return `LPPA_POLL_IO` when we know that an EAGAIN was previously detected during reading the proxy header. Otherwise, we should let the parent class' prepare method to decide about the next step (by publishing and calling `log_proto_text_server_prepare()`).

Reply:
Note: The transport's IO condition has priority over the proto's, so it has to be taken into account as well (see other proto implementations):

```
  *cond = self->super.transport->cond;

  /* if there's no pending I/O in the transport layer, then ... */
  if (*cond == 0)
```